Symfony/CS/Console/Command/FixCommand.php

```diff
@@ -434,10 +436,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
                         $output->write(sprintf($fixerDetailLine, implode(', ', $fixResult['appliedFixers'])));
                     }
 
-                    if ($input->getOption('diff')) {
+                    if ($isDiff) {
+                        $diff = OutputFormatter::escape($fixResult['diff']);
+
                         $output->writeln('');
                         $output->writeln('<comment>      ---------- begin diff ----------</comment>');
-                        $output->writeln($fixResult['diff']);
+
+                        if ($isDecorated) {
+                            $diff = implode(
+                                PHP_EOL,
+                                array_map(
+                                    function ($string) {
+                                        $string = preg_replace('/^(\+){3}/', '<info>+++</info>', $string);
+                                        $string = preg_replace('/^(\+){1}/', '<info>+</info>', $string);
+                                        $string = preg_replace('/^(\-){3}/', '<error>---</error>', $string);
+                                        $string = preg_replace('/^(\-){1}/', '<error>-</error>', $string);
+                                        $string = str_repeat(' ', 6).$string;
+
+                                        return $string;
+                                    },
+                                    explode(PHP_EOL, $diff)
+                                )
+                            );
+                        }
+
+                        $output->writeln($diff);
```

Comment:
still not sure how this could work for non-decorated consoles, the output does not remove escape characters (see line 440) when in non-decorated mode.